server/src/main/scala/com/cloudera/livy/LivyConf.scala

```diff
@@ -57,9 +58,22 @@ object LivyConf {
   val SUPERUSERS = Entry("livy.superusers", null)
 
   val AUTH_TYPE = Entry("livy.server.auth.type", null)
-  val KERBEROS_PRINCIPAL = Entry("livy.server.auth.kerberos.principal", null)
-  val KERBEROS_KEYTAB = Entry("livy.server.auth.kerberos.keytab", null)
-  val KERBEROS_NAME_RULES = Entry("livy.server.auth.kerberos.name_rules", "DEFAULT")
+  val AUTH_KERBEROS_PRINCIPAL = Entry("livy.server.auth.kerberos.principal", null)
+  val AUTH_KERBEROS_KEYTAB = Entry("livy.server.auth.kerberos.keytab", null)
+  val AUTH_KERBEROS_NAME_RULES = Entry("livy.server.auth.kerberos.name_rules", "DEFAULT")
+
+  val LAUNCH_KERBEROS_PRINCIPAL =
+    LivyConf.Entry("livy.server.launch.kerberos.principal", null)
+  val LAUNCH_KERBEROS_KEYTAB =
+    LivyConf.Entry("livy.server.launch.kerberos.keytab", null)
+  val LAUNCH_KERBEROS_REFRESH_INTERVAL =
+    LivyConf.Entry("livy.server.launch.kerberos.refresh_interval_seconds", 3600)
+  val KINIT_FAIL_THRESHOLD =
+    LivyConf.Entry("livy.server.launch.kerberos.kinit_fail_threshold", 5)
+  val KINIT_CACHE_ROOT_DIR =
+    LivyConf.Entry("livy.server.launch.kerberos.kinit_cache_root_dir",
+      Files.createTempDirectory("kinit_cache").toFile.getAbsolutePath)
```

Comment:
This will always create a temp directory, regardless of whether the user has set this or not. Better to do that only if the user has not set this. Also, "root" in the config name sounds a little confusing, it kinda implies there might be other subdirectories involved; I'd just remove it.